### PR TITLE
[Blueprint] Bug fix for publish and assignment

### LINKED ISSUE #1541: az blueprint assignment create - fails
## Describe the bug

**Command Name**
`az blueprint assignment create
Extension Name: blueprint. Version: 0.1.0.`

**Errors:**
```
'artifact_name'
Traceback (most recent call last):
python3.6/site-packages/knack/cli.py, ln 215, in invoke
    cmd_result = self.invocation.execute(args)
cli/core/commands/__init__.py, ln 625, in execute
    raise ex
cli/core/commands/__init__.py, ln 689, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
...
cliextensions/blueprint/azext_blueprint/custom.py, ln 526, in <dictcomp>
    body['resource_groups'] = {rg['artifact_name']: _del_artifact_name(rg) for rg in resource_groups} if resource_groups is not None else {}
KeyError: 'artifact_name'
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az blueprint assignment create --subscription {} --name {} --location {} --identity-type {} --description {} --blueprint-version {} --locks-mode {} --resource-group-value {} {} {} --parameters {}`

## Environment Summary
```
Linux-5.3.0-46-generic-x86_64-with-debian-buster-sid
Python 3.6.5
Installer: DEB

azure-cli 2.3.1

Extensions:
blueprint 0.1.0

```

### LINKED ISSUE #1598: publish is failing without change notes
az blueprint publish is failing when --change-notes are not supplied even though parameter is optional.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 82aa0874-eb79-9bed-f3fc-3a0e6d923c50
* Version Independent ID: 1971ba1e-c071-587c-2673-70d0a659cc32
* Content: [az blueprint](https://docs.microsoft.com/en-us/cli/azure/ext/blueprint/blueprint?view=azure-cli-latest#ext-blueprint-az-blueprint-publish)
* Content Source: [latest/docs-ref-autogen/ext/blueprint/blueprint.yml](https://github.com/MicrosoftDocs/azure-docs-cli/blob/live/latest/docs-ref-autogen/ext/blueprint/blueprint.yml)
* GitHub Login: @rloutlaw
* Microsoft Alias: **routlaw**

### PR DESCRIPTION
**Pull request by fengzhou-msft**
Fix #1598
Fix #1541

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Blueprint

**Comment by qwordy** [comment]
Could you summarize this PR with a few of sentences?

File: src/blueprint/azext_blueprint/custom.py [hunk 1]
```diff
@@ -480,8 +481,9 @@ def list_blueprint_version_artifact(cmd, client, blueprint_name,
 
 
 def _del_artifact_name(rg):
-    del rg['artifact_name']
-    return rg
+    rg_copy = rg.copy()
+    del rg_copy['artifact_name']
+    return rg_copy
```
Comment (qwordy) [inline src/blueprint/azext_blueprint/custom.py:486]: What's the story about this change?

Comment (fengzhou-msft) [reply src/blueprint/azext_blueprint/custom.py:486]: https://github.com/Azure/azure-cli-extensions/blob/30be073a9f3b954a7724fa841e218195c24cfb8b/src/blueprint/azext_blueprint/custom.py#L528

In Python 3.6 and 3.7 `_del_artifact_name(rg)` is executed before `rg['artifact_name']` while Python 3.8 has the expected order, so a key error is thrown in 3.6 and 3.7.

Comment (fengzhou-msft) [reply src/blueprint/azext_blueprint/custom.py:486]: It was a bug in Python 3.6 and 3.7 to evaluate values first in a dictionary comprehension and fixed in 3.8: https://stackoverflow.com/a/42203486/2100487

Comment (qwordy) [reply src/blueprint/azext_blueprint/custom.py:486]: Really tricky. I don't think it's a good practice to rely on the evaluation order of key and value. Maybe you can split it into multiple lines. `rg.copy()` has a extra overhead.

Comment (fengzhou-msft) [reply src/blueprint/azext_blueprint/custom.py]: Updated.
